### PR TITLE
check windows installer user rights assignment

### PR DESCRIPTION
**Pull request by clarkb7**
### What does this PR do?

Adds E2E test to check Windows Installer sets the user rights on the agent user.

### Motivation

https://datadoghq.atlassian.net/browse/WINA-507

### CODE REVIEW

File: test/new-e2e/tests/windows/install-test/assert.go [hunk 1]
```diff
@@ -42,3 +42,18 @@ func AssertInstalledUserInRegistry(t *testing.T, host *components.RemoteHost, ex
...
+		"SeServiceLogonRight",
+		"SeDenyInteractiveLogonRight",
+		"SeDenyNetworkLogonRight",
+		"SeDenyRemoteInteractiveLogonRight",
+	}
+	actualRights, err := windows.GetUserRightsForUser(host, username)
+	if !assert.NoError(t, err, "should get user rights") {
+		return false
+	}
+	return assert.ElementsMatch(t, expectedRights, actualRights, "user %s should have user rights", username)
```
Comment (iglendd) [inline test/new-e2e/tests/windows/install-test/assert.go:58]: Not sure ElementMatch logic. Will it succeed if there are more actual rights? Is that a problem in general and for future if it is?

File: test/new-e2e/tests/windows/install-test/installtester.go [hunk 2]
```diff
@@ -368,6 +368,12 @@ func (t *Tester) testCurrentVersionExpectations(tt *testing.T) {
 	require.NoError(tt, err)
 	serviceTester.TestInstall(tt)
 
+	tt.Run("user rights", func(tt *testing.T) {
+		AssertUserRights(tt, t.host,
+			windows.MakeDownLevelLogonName(t.expectedUserDomain, t.expectedUserName),
+		)
+	})
+
```
Comment (iglendd) [inline test/new-e2e/tests/windows/install-test/installtester.go:376]: Suggest in future PR move user checks is a separate block (e.g. https://github.com/DataDog/datadog-agent/pull/23419/files#diff-cf464d0b98afe8e5a03a4033f8584c6954d2eef118f6e471390c969b63436624R360-R376)

**Review by iglendd — APPROVED** [review]

**Review by richl-dd — APPROVED** [review]

**Comment by clarkb7** [comment]
rebasing for conflicts

**Comment by clarkb7** [comment]
rebase for ci fix

**Review by pducolin — APPROVED** [review]

File: test/new-e2e/tests/windows/install-test/installtester.go [hunk 3]
```diff
@@ -374,6 +374,12 @@ func (t *Tester) testCurrentVersionExpectations(tt *testing.T) {
 		)
 	})
 
+	tt.Run("user rights", func(tt *testing.T) {
```
[the remark this answers is not in the document]
Comment (clarkb7) [reply test/new-e2e/tests/windows/install-test/installtester.go:377]: it's to identify this particular test

**Pull request merged**